docs/kubernetes/windows-details.md

```diff
@@ -0,0 +1,335 @@
+
+```
+$ az vm image list --publisher MicrosoftWindowsServer --all -o table
+
+Offer                    Publisher                      Sku                                             Urn                                                                                                            Version
+-----------------------  -----------------------------  ----------------------------------------------  -------------------------------------------------------------------------------------------------------------  -----------------
+...
+WindowsServerSemiAnnual  MicrosoftWindowsServer         Datacenter-Core-1709-with-Containers-smalldisk  MicrosoftWindowsServer:WindowsServerSemiAnnual:Datacenter-Core-1709-with-Containers-smalldisk:1709.0.20180412  1709.0.20180412
+WindowsServerSemiAnnual  MicrosoftWindowsServer         Datacenter-Core-1803-with-Containers-smalldisk  MicrosoftWindowsServer:WindowsServerSemiAnnual:Datacenter-Core-1803-with-Containers-smalldisk:1803.0.20180504  1803.0.20180504
+```
+
+You can use the Offer, Publisher and Sku to pick a specific version by adding `windowsOffer`, `windowsPublisher`, `windowsSku` and (optionally) `widndowsVersion` to the `windowsProfile` section. In this example, the latest Windows Server version 1803 image would be deployed.
+
+```json
+"windowsProfile": {
+            "adminUsername": "azureuser",
+            "adminPassword": "...",
+            "windowsPublisher": "MicrosoftWindowsServer",
+            "windowsOffer": "WindowsServerSemiAnnual",
+            "windowsSku": "Datacenter-Core-1803-with-Containers-smalldisk"
+     },
+```
+
+
+
+## More Examples
+
+### Using Azure Files
+
+This example is modified after https://github.com/andyzhangx/Demo/tree/master/windows/azurefile/rs3
+
+1. Create an azure file storage class
+```kubectl apply -f https://raw.githubusercontent.com/JiangtianLi/Examples/master/windows/azurefile/storageclass-azurefile.yaml```
+
+2. Make sure storageclass is created successfully
+```
+kubectl get storageclass/azurefile -o wide
+```
+
+3. Create a pvc for azure file
+```kubectl apply -f https://raw.githubusercontent.com/JiangtianLi/Examples/master/windows/azurefile/pvc-azurefile.yaml```
+
+4. Make sure pvc is created successfully
+```
+kubectl get pvc/pvc-azurefile -o wide
+```
+
+5. Create a pod with azure file pvc
+```kubectl apply -f https://raw.githubusercontent.com/JiangtianLi/Examples/master/windows/azurefile/iis-azurefile.yaml```
+
+6. Watch the status of pod until its `STATUS` is `Running`
+```
+watch kubectl get po/iis-azurefile -o wide
+```
+
+7. Enter the pod container to validate
+```
+kubectl exec -it iis-azurefile -- cmd
+```
+
+```
+C:\>dir c:\mnt\azure
+ Volume in drive C has no label.
+ Volume Serial Number is F878-8D74
+
+ Directory of c:\mnt\azure
+
+11/16/2017  09:45 PM    <DIR>          .
+11/16/2017  09:45 PM    <DIR>          ..
+               0 File(s)              0 bytes
+               2 Dir(s)   5,368,709,120 bytes free
+
+```

```

Comment:
This might sound silly, but should we include examples for both Windows and Linux commands/file paths? I can definitely see scenarios where those on macOS / linux still might want to deploy a Windows K8s cluster.

Reply:
done